### ISSUE TITLE
/request/?view=collection - project filtering considers reviews

### ISSUE THREAD

**Issue by dmach** [openSUSE/open-build-service#13911]
# Issue Description
`osc api 'https://api.opensuse.org/request?view=collection&project=openSUSE:Tools&states=new,review,declined'`
contains
```
<request id="1065089" creator="maxlin_factory">
  <action type="submit">
    <source project="openSUSE:Factory" package="openSUSE-release-tools" rev="458"/>
    <target project="openSUSE:Backports:SLE-15-SP5" package="openSUSE-release-tools"/>
  </action>
  <review state="new" when="2023-02-13T06:01:08" who="factory-auto" by_package="openSUSE-release-tools" by_project="openSUSE:Tools">
    <comment>Submission for openSUSE-release-tools by someone who is not maintainer in the devel project (openSUSE:Tools). Please review</comment>
  </review>
```
Please note that the request has no source or target in 'openSUSE:Tools' but there's an open review for this project.

Expected Result
============
Reviewers should not be considered when filtering the requests.
If needed, a new search field for such cases should be added, for example `reviewed_by_project`.

**Comment by eduardoj** [comment]
After having a look a the code, I realized that the  `project` parameter of the `/request?view=collection` API endpoint was designed to search by these three fields:
- [source projects](https://github.com/openSUSE/open-build-service/blob/d258e420309410e1e2cb7f1bae741eb0140b0db5/src/api/app/models/bs_request/find_for/project.rb#L6),
- [target projects](https://github.com/openSUSE/open-build-service/blob/d258e420309410e1e2cb7f1bae741eb0140b0db5/src/api/app/models/bs_request/find_for/project.rb#L7), or 
- [review by project](https://github.com/openSUSE/open-build-service/blob/d258e420309410e1e2cb7f1bae741eb0140b0db5/src/api/app/models/bs_request/find_for/project.rb#L14-L18)

The functionality you are requiring (searching only by source, or target project, or by both) can be achieved by the  `/search/request?match=...` API endpoint. Could you consider using the `/search/request?match=...` API endpoint for your needs.

Removing the "Bug" label.

**Comment by dmach** [comment]
@eduardoj thanks for looking into this.
This explains a lot. So the `/request?view=collection` has a special purpose and I should really consider moving back to the previous API call.

Could you fix API docs accordingly? I find the current description misleading:
```
GET /request
  project: limit to result to defined target project or review requests
  package: limit to result to defined target package or review requests 
```
It might be also good to explain what's the purpose of the endpoint and that users should use the other one otherwise.

**Comment by eduardoj** [comment]
> @eduardoj thanks for looking into this. This explains a lot. So the `/request?view=collection` has a special purpose and I should really consider moving back to the previous API call.
> 
> Could you fix API docs accordingly? I find the current description misleading:
> 
> ```
> GET /request
>   project: limit to result to defined target project or review requests
>   package: limit to result to defined target package or review requests 
> ```

I agree. I have just created #13915, to complete the current work in progress of apidocs-new. Feel free to review that pull request, and add your suggestions there.

I'm not against of modifying the current `/apidocs`, but adapting that document to provide something similar to what apidocs-new already achieves, looks too much to me.

> It might be also good to explain what's the purpose of the endpoint and that users should use the other one otherwise.

What the endpoint returns is described in [these lines](https://github.com/openSUSE/open-build-service/blob/d258e420309410e1e2cb7f1bae741eb0140b0db5/src/api/public/apidocs-new/paths/request_view_collection.yaml#L2-L13), and can be browsed here: https://build.opensuse.org/apidocs-new/#/Requests/get_request_view_collection

It looks like this, now:

![Screenshot from 2023-02-28 14-59-28](https://user-images.githubusercontent.com/24919/221876664-e7a1740a-77ef-42eb-a358-bf1938520e8a.png)

Regarging "... and that users should use the other one otherwise", yes, the API reference documentation could point to other related endpoints. But giving hints about which endpoint to use depending on specific use cases belongs to an independent API user documentation which is not yet developed, and should not make part of the API reference documentation (done with the OpenAPI specification), IMHO.
